### PR TITLE
24 pull requests (Independent Publisher)

### PR DESCRIPTION
**Pull request by Bkaraba**
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [x] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [x] I attest that the connector works and I verified by deploying and testing all the operations. 
- [x] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [x] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [x] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [x] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [x] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [x] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [x] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [x] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

### CODE REVIEW

**Comment by Bkaraba** [comment]
Screenshot from 2022-10-07 12-57-52
Screenshot from 2022-10-07 12-58-15
Screenshot from 2022-10-07 12-58-33
Screenshot from 2022-10-07 12-58-56

Screenshot from 2022-10-07 12-59-28
Screenshot from 2022-10-07 13-00-31
Screenshot from 2022-10-07 13-01-03

File: independent-publisher-connectors/Proposal - 24 Pull Request (Independent Publisher)/apiDefinition.swagger.json [hunk 1]
```diff
@@ -0,0 +1,262 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "24 pull request",
+    "description": "connects to '24 pull request' api for open source projects contibution",
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Proposal - 24 Pull Request (Independent Publisher)/apiDefinition.swagger.json:5]: Hi @Bkaraba,
Thank you so much for submitting your connector proposal.
I wanted to bring your attention to some points you may need to fix.
Please remove the API words from the description. Please follow [these requirements](https://learn.microsoft.com/en-us/connectors/custom-connectors/certification-submission#write-a-description-for-your-connector) for the description.

File: independent-publisher-connectors/Proposal - 24 Pull Request (Independent Publisher)/apiDefinition.swagger.json [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+    "version": "1.0",
+    "contact": {
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Proposal - 24 Pull Request (Independent Publisher)/apiDefinition.swagger.json:7]: The contact section should include a "url" : "the product website url"
Please see our template [here](https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/Independent%20Publisher/apiDefinition.swagger.json)

File: independent-publisher-connectors/Proposal - 24 Pull Request (Independent Publisher)/apiDefinition.swagger.json [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+  "paths": {
+    "/users.json": {
+      "get": {
+        "responses": {
+          "default": {
+            "description": "default",
+            "schema": {}
+          }
+        },
+        "summary": "get users",
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Proposal - 24 Pull Request (Independent Publisher)/apiDefinition.swagger.json:28]: Please capitalize the first letter.

File: independent-publisher-connectors/Proposal - 24 Pull Request (Independent Publisher)/apiDefinition.swagger.json [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+        "description": "get all users",
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Proposal - 24 Pull Request (Independent Publisher)/apiDefinition.swagger.json:29]: all Operations descriptions should end with a period. No period required only if there's a single word.
Please see this useful [documentation](https://learn.microsoft.com/en-us/connectors/custom-connectors/connector-string-guidance#operations)

File: independent-publisher-connectors/Proposal - 24 Pull Request (Independent Publisher)/apiDefinition.swagger.json [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+        "operationId": "GetUsers",
+        "parameters": [
+          {
+            "name": "page",
+            "in": "query",
+            "required": false,
+            "type": "integer",
+            "description": "the page number to display",
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Proposal - 24 Pull Request (Independent Publisher)/apiDefinition.swagger.json:37]: all Parameters descriptions should end with a period. No period required only if there's a single word.

File: independent-publisher-connectors/Proposal - 24 Pull Request (Independent Publisher)/apiDefinition.swagger.json [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+  "basePath": "/",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [],
+  "produces": [],
+  "paths": {
+    "/users.json": {
+      "get": {
+        "responses": {
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Proposal - 24 Pull Request (Independent Publisher)/apiDefinition.swagger.json:22]: default responses in not allowed, please define a proper response for this action.

File: independent-publisher-connectors/Proposal - 24 Pull Request (Independent Publisher)/apiDefinition.swagger.json [hunk 7]
```diff
@@ -0,0 +1,262 @@
...
+          "default": {
+            "description": "default",
+            "schema": {}
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Proposal - 24 Pull Request (Independent Publisher)/apiDefinition.swagger.json:25]: empty schema is not allowed. please define a proper schema.

File: independent-publisher-connectors/Proposal - jenga (Independent Publisher)/README.md [hunk 8]
```diff
@@ -0,0 +1,3 @@
+# JENGA
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Proposal - jenga (Independent Publisher)/README.md:1]: Hi @Bkaraba,
Thank you so much for your hard work with us.
If you would like us to move forward with the "24 pull requests" connector, please remove the other Proposal "Jenga" file.
Please let us know if you have any questions.

**Comment by Bkaraba** [comment]
@microsoft-github-policy-service agree company="no company specified"

File: independent-publisher-connectors/Proposal - 24 Pull Request (Independent Publisher)/apiDefinition.swagger.json (hunk 2 above)
Comment (Bkaraba) [reply independent-publisher-connectors/Proposal - 24 Pull Request (Independent Publisher)/apiDefinition.swagger.json:7]: fixed

File: independent-publisher-connectors/Proposal - 24 Pull Request (Independent Publisher)/apiDefinition.swagger.json (hunk 6 above)
Comment (Bkaraba) [reply independent-publisher-connectors/Proposal - 24 Pull Request (Independent Publisher)/apiDefinition.swagger.json:23]: fixed

File: independent-publisher-connectors/Proposal - jenga (Independent Publisher)/README.md (hunk 8 above)
Comment (Bkaraba) [reply independent-publisher-connectors/Proposal - jenga (Independent Publisher)/README.md]: fixed

**Comment by Amjed-Ayoub** [comment]
Hi @Bkaraba, May I ask you please to resolve all the conversations.

**Comment by Amjed-Ayoub** [comment]
Hi @Bkaraba May I ask you please to change the folder name and remove the word Proposal from it. 
thank you so much

**Comment by Amjed-Ayoub** [comment]
@Bkaraba thank you so much for all your hard work on this. May I ask you again please to remove the (independent publisher) from the folder name as well. the folder name should contain your connector name only.
Thank you so much again.

**Comment by Amjed-Ayoub** [comment]
[[certify-connector]]

**Comment by Amjed-Ayoub** [comment]
Hello @Bkaraba,
I hope you are doing well.
Congratulations, your pull request is approved. We will move forward with the certification process. We are preparing your connector for production deployment and will queue up for the next deployment schedule. Once your connector onboards the next deployment schedule, it starts to deploy your connector in our production environments that typically takes 3-4 weeks. 
Please let us know if you have any questions.
Thank you very much for working with us.

**Comment by Amjed-Ayoub** [comment]
Hello @Bkaraba,
I hope you are doing well.
Your connector is in production deployment and queued up for the next deployment schedule. This process typically takes 3-4 weeks. Please note that the documentation for your connector will not be generated until the deployment process is in the final steps.
Please let us know if you have any questions.
Thank you very much for working with us.

**Comment by Amjed-Ayoub** [comment]
Hello @Bkaraba,
I hope you are doing well.
Congratulations, the deployment of your connector and documentation page was already completed, We have a short [survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbR0BHMrjhL0hDmckROosW6AFURERIVUUzMk5XRzJGV1dJM1Q5RVZJUjFXSC4u) on the connector certification process that we would like for you to complete. Your feedback is appreciated and will help us improve the program.
Thank you very much for working with us.

**Review by Amjed-Ayoub — APPROVED** [review]

**Pull request merged**